client/client_test.go

```diff
@@ -68,23 +70,24 @@ func TestAddLeaf(t *testing.T) {
 		},
 	} {
 		client := New(logID, test.client, testonly.Hasher, env.PublicKey)
-		client.SetMaxTries(1)
-
-		if err, want := client.AddLeaf(ctx, []byte(test.desc)), codes.DeadlineExceeded; grpc.Code(err) != want {
-			t.Errorf("AddLeaf(%v): %v, want, %v", test.desc, err, want)
-			continue
+		{
+			ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(timeout))
```

Comment:
Is this liable to cause test flakes by use of absolute time? Can we use a mock context somehow if so?